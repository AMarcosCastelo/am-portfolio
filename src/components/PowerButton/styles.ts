import styled from 'styled-components'

export const Power = styled.button`
  position: fixed;
  top: 2rem;
  left: 50%;
  transform: translate(-50%, 0);

  background-color: ${({ theme }) => theme.palette.background.default};
  padding: 0.3rem;
  border-radius: 50%;
  border: 1px solid #000;
  width: 2.5rem;
  height: 2.5rem;
`
