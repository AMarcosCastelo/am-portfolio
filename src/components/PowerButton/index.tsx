import NextLink from 'next/link'
import { PowerBtn } from 'components/AllSvgs'

import * as S from './styles'

const PowerButton = () => (
  <S.Power>
    <NextLink href="/">
      <a>
        <PowerBtn width={30} height={30} fill="currentColor" />
      </a>
    </NextLink>
  </S.Power>
)

export default PowerButton
