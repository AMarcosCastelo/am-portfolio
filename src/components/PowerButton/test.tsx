import { render, screen } from '@testing-library/react'

import PowerButton from '.'

describe('<PowerButton />', () => {
  it('should render the heading', () => {
    const { container } = render(<PowerButton />)

    expect(
      screen.getByRole('heading', { name: /PowerButton/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
