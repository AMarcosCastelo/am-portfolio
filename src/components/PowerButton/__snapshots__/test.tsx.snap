// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PowerButton /> should render the heading 1`] = `
.c0 {
  position: fixed;
  top: 2rem;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  background-color: #FCF6F4;
  padding: 0.3rem;
  border-radius: 50%;
  border: 1px solid #000;
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 3;
  cursor: pointer;
}

.c0:hover {
  background-color: rgba(0,255,0,0.4);
  box-shadow: 0 0 8px 6px rgba(0,255,0,0.2);
}

.c0:hover > *:first-child {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

<button
  class="c0"
>
  <a
    href="/"
  >
    <svg
      aria-hidden="true"
      class="prefix__svg-inline--fa prefix__fa-power-off prefix__fa-w-16"
      data-icon="power-off"
      data-prefix="fas"
      fill="currentColor"
      height="30"
      viewBox="0 0 512 512"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 54.1c63 45 104 118.6 104 201.9 0 136.8-110.8 247.7-247.5 248C120 504.3 8.2 393 8 256.4 7.9 173.1 48.9 99.3 111.8 54.2c11.7-8.3 28-4.8 35 7.7L162.6 90c5.9 10.5 3.1 23.8-6.6 31-41.5 30.8-68 79.6-68 134.9-.1 92.3 74.5 168.1 168 168.1 91.6 0 168.6-74.2 168-169.1-.3-51.8-24.7-101.8-68.1-134-9.7-7.2-12.4-20.5-6.5-30.9l15.8-28.1c7-12.4 23.2-16.1 34.8-7.8zM296 264V24c0-13.3-10.7-24-24-24h-32c-13.3 0-24 10.7-24 24v240c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24z"
        fill="currentColor"
      />
    </svg>
  </a>
</button>
`;
